test/test_get.py

```diff
@@ -0,0 +1,146 @@
+import unittest2
+from test.helpers.httpretty_extension import httpretty
+from test.helpers.merged_datasets_helper import setupDatasetsTest
+import pandas
+import numpy
+from mock import patch, call
+from quandl.model.dataset import Dataset
+from quandl.model.merged_dataset import MergedDataset
+from quandl.get import get
+from quandl.api_config import ApiConfig
+
+
+class GetSingleDatasetTest(unittest2.TestCase):
+    @classmethod
+    def setUp(self):
+        # ensure api key is not set
+        ApiConfig.api_key = None
+        setupDatasetsTest(self, httpretty)
+
+    @classmethod
+    def tearDownClass(cls):
+        httpretty.disable()
+        httpretty.reset()
+
+    def test_returns_pandas_by_default(self):
+        result = get('NSE/OIL')
+        self.assertIsInstance(result, pandas.core.frame.DataFrame)
+
+    def test_returns_pandas_when_requested(self):
+        result = get('NSE/OIL', returns='pandas')
+        self.assertIsInstance(result, pandas.core.frame.DataFrame)
+
+    def test_returns_numpys_when_requested(self):
+        result = get('NSE/OIL', returns='numpy')
+        self.assertIsInstance(result, numpy.core.records.recarray)
+
+    def test_setting_api_key_config(self):
+        ApiConfig.api_key = 'api_key_configured'
+        get('NSE/OIL')
+        self.assertEqual(ApiConfig.api_key, 'api_key_configured')
```

Comment:
I think for this test we probably want to check that the outgoing url includes the api key rather than the fact that the key we just set was infact set.